File: entity/src/main/java/far/galaxy/foodcourt/entity/customer/Customer.java
package far.galaxy.foodcourt.entity.customer;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import org.springframework.lang.NonNull;

import javax.persistence.*;
import java.math.BigDecimal;

@Entity
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Customer {
    private long id;
    private String name;
    private String email;
    private BigDecimal balance;

    public Customer() {
        this.id = 0;
        this.balance = BigDecimal.valueOf(0);
    }

    public Customer(String name, String email) {
        this();
        this.name = name;
        this.email = email;
    }

    @Id
    @NonNull
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    @NonNull
    @Column(unique = true)
    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    @NonNull
    @Column(unique = true)
    @JsonIgnore
    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public BigDecimal getBalance() {
        return balance;
    }

    public void setBalance(BigDecimal balance) {
        this.balance = balance;
    }

    public void addBalance(BigDecimal amount) {
        this.balance = this.balance.add(amount);
    }

    public void minusBalance(BigDecimal amount) {
        this.balance = this.balance.subtract(amount);
    }
}

File: white/src/main/javascript/far/galaxy/foodcourt/web/entity/Customer.js
import ChangeableEntity from "./ChangeableEntity";

export default class Customer extends ChangeableEntity {
    constructor(value) {
        super(value);
    }

    getId() {
        return this.value.id;
    }

    getName() {
        return this.value.name || "";
    }

    setName(name) {
        this.markAsChanged();
        this.value.name = name;
    }

    getEmail() {
        return this.value.email || "";
    }

    setEmail(email) {
        this.markAsChanged();
        this.value.email = email;
    }

    getBalance() {
        return this.value.balance || 0;
    }
}
